metadata-ingestion/src/datahub/ingestion/source/powerbi_report_server/report_server.py

```diff
@@ -0,0 +1,577 @@
+#########################################################
+#
+# Meta Data Ingestion From the Power BI Report Server
+#
+#########################################################
+import logging
+from dataclasses import dataclass
+from dataclasses import field as dataclass_field
+from typing import Any, Dict, Iterable, List, Optional, Set
+
+import requests
+from orderedset import OrderedSet
+from pydantic import ValidationError
+from pydantic.fields import Field
+from requests.exceptions import ConnectionError
+from requests_ntlm import HttpNtlmAuth
+
+import datahub.emitter.mce_builder as builder
+from datahub.configuration.common import AllowDenyPattern
+from datahub.configuration.source_common import EnvBasedSourceConfigBase
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.decorators import (
+    SourceCapability,
+    SupportStatus,
+    capability,
+    config_class,
+    platform_name,
+    support_status,
+)
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.common import ChangeAuditStamps
+from datahub.metadata.schema_classes import (
+    BrowsePathsClass,
+    ChangeTypeClass,
+    CorpUserInfoClass,
+    CorpUserKeyClass,
+    DashboardInfoClass,
+    DashboardKeyClass,
+    OwnerClass,
+    OwnershipClass,
+    OwnershipTypeClass,
+    StatusClass,
+)
+
+from .constants import API_ENDPOINTS, Constant
+from .graphql_domain import CorpUser
+from .report_server_domain import LinkedReport, MobileReport, PowerBiReport, Report
+
+LOGGER = logging.getLogger(__name__)
+
+
+class PowerBiReportServerAPIConfig(EnvBasedSourceConfigBase):
+    username: str = Field(description="Windows account username")
```

Comment:
Could you please use pydantic instead of Field, Please check https://github.com/datahub-project/datahub/blob/3aa74e0431a1d69c1ef1d4a7bbd61849f62c3121/metadata-ingestion/src/datahub/ingestion/source/powerbi.py#L108

Reply:
> Could you please use pydantic instead of Field, Please check
> 
> https://github.com/datahub-project/datahub/blob/3aa74e0431a1d69c1ef1d4a7bbd61849f62c3121/metadata-ingestion/src/datahub/ingestion/source/powerbi.py#L108

refactored